src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -26,6 +28,10 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
     readonly templateSrv: TemplateSrv = getTemplateSrv()
   ) {
     super(instanceSettings, backendSrv, templateSrv);
+
+    this.querySystems().then(
+      systems => this.state.systems = systems
+    ).catch( _ => { return;});
```

Comment:
We should have floating errors . Please check @mure comment :

https://github.com/ni/systemlink-grafana-plugins/pull/69#discussion_r1781892686

Reply:
```
 <FloatingError message={errorMsg}/> 
```

Reply:
We also need to ensure we are falling back to default name in case this systems can't be queried

Reply:
I feel like I have to share the reason behind `catch( _ => { return; });`. 
I'm not sure how to mock this in test to not spam errors. 

I think that the best solution is to have a function that checks if the system is set when the runQuery is called.

Reply:
I like this code more. Now we will query for the system when the user first runs a query. If we encounter an error then we display the IDs and we will retry to obtain the system the next time the user runs a query. 

If the system requests success we keep it in memory.